test/drop_test/action.py

```diff
@@ -261,6 +261,30 @@ def __str__(self):
         return f"Wait({str(self._event)})"
 
 
+class WaitForOneOf(Action):
+    def __init__(self, events: typing.List[Event]):
+        self._events: typing.List[Event] = events
+        self._found_event = False
+
+    async def run(self, drop: ffi.Drop):
+        e = await drop._events.wait_for_any_event(100, ignore_progress=True)
+
+        if e is None:
+            raise Exception(f"Expected one of {self._events} but got nothing")
+
+        if e not in self._events:
+            raise Exception(f"Expected one of {self._events} but got {e}")
+        else:
+            if self._found_event:
```

Comment:
Is this flag needed?

Reply:
ah you're right, this only runs once :facepalm: